src/map/packet_system.cpp

```diff
@@ -955,6 +955,12 @@ void SmallPacket0x01A(map_session_data_t* const PSession, CCharEntity* const PCh
         break;
         case 0x07: // weaponskill
         {
+            if (!PChar->PAI->IsEngaged()) // Prevent Weaponskill usage if player isn't engaged.
```

Comment:
This isnt retail behavior, SE allows you to do this exact thing via packet injection. Don't know if we really care though

Reply:
I assume that if something can only be pulled off via packet injection, it's unintended behavior that would fall into "better than retail"

Reply:
I don't think it does block you at all unless they changed it recently, shortcuts + gearswap and typing out the name through `//weaponskill` injects an outgoing 0x01A, as well as you can make the argument that ranged weapon skills shouldn't need you to engage to weaponskill. I presume shorthand + luashitacast will do the same

this has been retail behavior since at least abyssea, people were able to do this with spellcast. perhaps this is a good candidate for a setting

Reply:
Seconding the idea of putting this behind a setting.  Reduces the risk overall of trying to hunt down any side effect that might happen from this.

Reply:
retail only "allows" it by injection or memory patching (certain add-ons will make this happen), because the devs trusted the client too much when the game was made.

setting third'd

add-on is likely why you've not seen it block you winter.

Reply:
Added a setting to this, so it can be easily toggled on/off

For retail accuracy, default is "off"